src/manager.cpp

```diff
@@ -376,8 +376,29 @@ void Manager::collectSingleFruVpd(
 {
     try
     {
-        // TODO: Check if CM flag enabled for the given D-bus object path in
-        // system config JSON.
+        // Get system config JSON object from worker class
+        nlohmann::json l_sysCfgJsonObj{};
+
+        if (m_worker.get() != nullptr)
+        {
+            l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+        }
+
+        // Check if system config JSON is present
+        if (l_sysCfgJsonObj.empty())
+        {
+            throw std::runtime_error(
+                "System config JSON object not present. Single FRU VPD collection failed for " +
+                std::string(i_dbusObjPath));
+        }
+
+        if (!jsonUtility::isVPDRecollectableAtRuntime(l_sysCfgJsonObj,
```

Comment:
With this check we are limiting this API for FRUs which are recollectable at standby, Could you please check if we make it generic so that it can be used in other case as well?
Like performVpdRecollection? That way we might not have to duplicate the code again.

Reply:
modified the code, so that manager::collectSingleFruVpd() api can be used in common by other apis.